magento/magento2
# Use two-way value binding for date field

### Description (*)


The previous `attr` binding would correctly set the initial value, but the `value` binding is required to make the field value update when the `shiftedValue` observable changes in the date component.

The problem that inspired this PR can be shown via the date filter in the admin order grid. Prior to this PR, clearing a date filter did correctly clear the filter, but did not reset the values of the "From" and "To" fields in the filtering toolbar. 

https://user-images.githubusercontent.com/2262772/104122030-4928df00-534b-11eb-87e9-2ea97bb661a9.mp4

This caused a discrepancy in the filter toolbar and and the actual filter state. Now clicking "Apply Filters" doesn't do anything, since in the filter state there are no filters, even though the date filter "From" and "To" filter fields have values set.

https://user-images.githubusercontent.com/2262772/104122083-8b522080-534b-11eb-99f4-cd73d755c2a1.mp4

Replacing the `attr` binding for value with KO's two-way `value` binding fixes the issue, since now the date field values are kept in sync with the filter state.

https://user-images.githubusercontent.com/2262772/104122111-c81e1780-534b-11eb-827c-5254930243bc.mp4

### Manual testing scenarios (*)


1. Log in to the Magento backend.
2. Go to Sales > Orders
3. Set a Purchase Date filter and click "Apply Filters"
4. Clear the Purchase Date filter from the active filters list (by clicking the small cross icon).
5. Notice that even though the Purchase Date filter was cleared, the filter values in date fields were in fact not, even though they should be cleared as well (just like the normal text / price field filters are).

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#31648: Use two-way value binding for date field
